### PR TITLE
Roll back uncommitted (and not-rolled-back) read/write transactions on session release

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs [hunk 1]
```diff
@@ -233,6 +234,20 @@ private void DeleteSessionFireAndForget(PooledSession session)
             ConsumeBackgroundTask(DeleteSessionAsync(session), "session deletion");
         }
 
+        /// <summary>
+        /// Rolls back the specified transaction, if any, in a fire-and-forget manner.
+        /// </summary>
+        /// <param name="session">The name of the session that owns the transaction.</param>
+        /// <param name="transactionID">The ID of the transaction to rollback. May be null, in which case this method is a no-op.</param>
+        private void MaybeRollbackTransactionFireAndForget(SessionName session, ByteString transactionID)
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs:242]: I think this is not called from anywhere anymore.

File: apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs [hunk 2]
```diff
@@ -244,5 +259,18 @@ private async Task DeleteSessionAsync(PooledSession session)
                 _logger.Warn("Failed to delete session. Session will be abandoned to garbage collection.", e);
             }
         }
+
+        private async Task RollbackTransactionAsync(SessionName sessionName, ByteString transactionId)
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs:263]: Same here, not called from anywhere?

File: apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs:242]: Doh! Absolutely. Will get rid of both.

**Comment by jskeet** [comment]
The Travis failure is unrelated - it's because a release is in progress.

**Pull request merged**